frappe/public/js/frappe/ui/toolbar/toolbar.js

```diff
@@ -230,6 +245,15 @@ frappe.ui.toolbar.clear_cache = frappe.utils.throttle(function () {
 	});
 }, 10000);
 
+frappe.ui.toolbar.reset_announcement_widget = function () {
+	frappe.db.get_single_value("Navbar Settings", "announcement_widget").then((value) => {
```

Comment:
This will cause a new request with every desk load? :eyes:

Reply:
In general sending small global values in boot is fine even though we villify boot. Boot is huge not because of these small global flag/values, it's because of translations and other such large data.

Reply:
> We can send this with boot, will also eliminate need to check if it's empty etc on client side code.

Had tried to store it in boot previously, but every change in announcements will then require a hard refresh so that the new value for it gets updated in the boot obj. Didn't know if that should be the case especially with things like announcements so used this instead. Can make changes to use boot instead ig.

Reply:
Hard refresh shouldn't be required. Boot is cached but there are two stages of building boot. 2nd stage is not cached and is called on every desk load. See app.py and trace boot creation from there.